File: day2/go/main.go
package main

import "sort"

func topKFrequent(nums []int, k int) []int {
    freq := map[int]int{}
    for _, n := range nums {
        freq[n]++
    }
    type kv struct {
        Key int
        Val int
    }
    pairs := []kv{}
    for key, val := range freq {
        pairs = append(pairs, kv{key, val})
    }
    sort.Slice(pairs, func(i, j int) bool {
        return pairs[i].Val > pairs[j].Val
    })
    res := []int{}
    for i := 0; i < k; i++ {
        res = append(res, pairs[i].Key)
    }
    return res
}

File: day2/typescript/main.ts
export function topKFrequent(nums: number[], k: number): number[] {
    const freq: Record<number, number> = {};
    nums.forEach(n => freq[n] = (freq[n] || 0) + 1);
    return Object.entries(freq)
        .sort((a, b) => b[1] - a[1])
        .slice(0, k)
        .map(entry => parseInt(entry[0]));
}
